Foreman 3.16 Release Notes

ogajduse opened the pull request:
#### What changes are you introducing?
Adding 3.16 release notes

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

#### Contributor checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.16/Katello 4.18 (Satellite 6.18)
* [ ] Foreman 3.15/Katello 4.17
* [ ] Foreman 3.14/Katello 4.16 (Satellite 6.17)
* [ ] Foreman 3.13/Katello 4.15 (EL9 only)
* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16; orcharhino 7.2 on EL9 only; orcharhino 7.3)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only; orcharhino 7.0 on EL8+EL9; orcharhino 7.1 with Leapp)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* We do not accept PRs for Foreman older than 3.9.

#### Review checklists

Tech review (performed by an Engineer who did not author the PR; can be skipped if tech review is unnecessary):

* [ ] The PR documents a recommended, user-friendly path.
* [ ] The PR removes steps that have been made unnecessary or obsolete.
* [ ] Any steps introduced or updated in the PR have been tested to confirm that they lead to the documented end result.

Style review (by a Technical Writer who did not author the PR):

* [ ] The PR conforms with the team's style guidelines.
* [ ] The PR introduces documentation that describes a user story rather than a product feature.

ogajduse commented:
Hammer release notes added. 

I am waiting for https://github.com/theforeman/theforeman.org/pull/2241 to be merged to `Make sure headline features, upgrade warnings and deprecations are in sync with the website`.

fraenki commented:
@ogajduse, would you please add a note regarding the new oVirt plugin? This was discussed [here](https://community.theforeman.org/t/rfc-deprecation-and-removal-of-ovirt-compute-resource/42527/35?u=fraenki). Users of the oVirt compute resource have to install the new plugin, otherwise their oVirt data is removed from Foreman 3.16. Thanks.

maximiliankolb commented on guides/doc-Release_Notes/topics/foreman-3.16.0.adoc:
```diff
@@ -0,0 +1,162 @@
+
+* pass:[Support setting key-algorithm for ISC DHCP] - https://projects.theforeman.org/issues/36325[#36325]
+
+=== Hammer CLI
```
```suggestion
=== Hammer CLI

```

maximiliankolb commented on guides/doc-Release_Notes/topics/foreman-3.16.0.adoc:
```diff
@@ -0,0 +1,162 @@
+=== Hammer CLI
+* pass:[Update field documentation to reflect replaced_by translation] http://projects.theforeman.org/issues/38427[#38427]
+
+=== Hammer CLI - Foreman
```
```suggestion
=== Hammer CLI - Foreman

```

ogajduse commented:
I've addressed all review comments as well as I added Upgrade Warnings, Headline Features, and Deprecations to streamline the release notes with the RN we have on https://theforeman.org/manuals/3.16/index.html#Releasenotesfor3.16.

ogajduse commented:
> @ogajduse, would you please add a note regarding the new oVirt plugin? This was discussed [here](https://community.theforeman.org/t/rfc-deprecation-and-removal-of-ovirt-compute-resource/42527/35?u=fraenki). Users of the oVirt compute resource have to install the new plugin, otherwise their oVirt data is removed from Foreman 3.16. Thanks.

@fraenki can you please check the release notes? I've added it to the `Upgrade Warnings` section.

fraenki commented on guides/doc-Release_Notes/topics/foreman.adoc:
```diff
@@ -15,10 +26,25 @@ For more information, see {AdministeringDocURL}limiting-host-resources[Limiting
+* *Previous Deprecation:* Feature was deprecated in {Project} 7.2
+endif::[]
+* *Impact:* Users with RHV/oVirt compute resources will lose this functionality unless they install the new ForemanOvirt plugin
+* *Alternative Solution:* Install the new ForemanOvirt plugin to maintain oVirt functionality
+* *Action Required:* If using RHV compute resources, plan migration to alternative platforms or install the ForemanOvirt plugin before upgrading. Users who want to keep oVirt support must install the ForemanOvirt plugin before upgrading. Users who do not want oVirt support are advised to run `foreman-rake ovirt:drop` before upgrading.
```
@ogajduse LGTM :+1: Thank you!

Lennonka reviewed (approved):
LGTM, thank you!
